File: euler002.cs
//Project Euler #2: Even Fibonacci numbers
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {

    static void Main(String[] args) {
        int t = Convert.ToInt32(Console.ReadLine());
        for(int a0 = 0; a0 < t; a0++){
            long n = Convert.ToInt64(Console.ReadLine());
            long a=0,b=2;
            long sum = a + b;
            while(b<=n)
            {
                long c = (4 *b) + a;
                if(c>=n)
                    break;
                a=b;
                b=c;
                sum += c;
            }
        Console.WriteLine(sum);
        }
    }
}

File: Euler002.java
//Project Euler #2: Even Fibonacci numbers
import java.util.Scanner;

public class Euler002 {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            long n = in.nextLong();
            long a=0,b=2;
            long c = 4 * b + a;
            long sum = a + b;
            do
            {                
                sum += c;              
                a=b;
                b=c;
                c = 4 * b + a;
            }
            while(c<=n);
        System.out.println(sum);
        }
        in.close();
    }
}
